sdk/storage/Azure.Storage.DataMovement.Files.Shares/src/ShareFileStorageResource.cs

```diff
@@ -69,13 +71,16 @@ internal class ShareFileStorageResource : StorageResourceItemInternal
                     throw Errors.ShareFileAlreadyExists(ShareFileClient.Path);
                 }
             }
+            ShareFileHttpHeaders httpHeaders = _options?.GetShareFileHttpHeaders(properties?.RawProperties);
+            IDictionary<string, string> metadata = _options?.GetFileMetadata(properties?.RawProperties);
+            FileSmbProperties smbProperties = _options?.GetFileSmbProperties(properties);
             await ShareFileClient.CreateAsync(
                     maxSize: maxSize,
-                    httpHeaders: _options.HttpHeaders,
-                    metadata: _options.FileMetadata,
-                    smbProperties: _options.SmbProperties,
-                    filePermission: _options.FilePermissions,
-                    conditions: _options.DestinationConditions,
+                    httpHeaders: _options?.GetShareFileHttpHeaders(properties?.RawProperties),
+                    metadata: _options?.GetFileMetadata(properties?.RawProperties),
+                    smbProperties: _options?.GetFileSmbProperties(properties),
```

Comment:
Did you mean to use the variables you created above here?

Reply:
Ah I moved it outside of it cause I was trying to debug something. But yes I did mean that. Good catch.

Reply:
done